WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationAccessUpdateEventProcessor.swift

```diff
@@ -26,15 +26,49 @@ protocol ConversationAccessUpdateEventProcessorProtocol {
     ///
     /// - Parameter event: A conversation access update event.
 
-    func processEvent(_ event: ConversationAccessUpdateEvent) async throws
+    func processEvent(_ event: ConversationAccessUpdateEvent) async
 
 }
 
 struct ConversationAccessUpdateEventProcessor: ConversationAccessUpdateEventProcessorProtocol {
 
-    func processEvent(_: ConversationAccessUpdateEvent) async throws {
-        // TODO: [WPB-10164]
-        assertionFailure("not implemented yet")
+    let repository: any ConversationRepositoryProtocol
+    let localStore: any ConversationLocalStoreProtocol
+
+    func processEvent(_ event: ConversationAccessUpdateEvent) async {
+        let conversationID = event.conversationID
+
+        let localConversation = await repository.fetchOrCreateConversation(
+            with: conversationID.uuid,
+            domain: conversationID.domain
+        )
+
+        let accessRoles = if let legacyAccessRole = event.legacyAccessRole {
+            getAccessRoles(from: legacyAccessRole)
+        } else {
+            event.accessRoles ?? [.teamMember, .nonTeamMember, .service]
+        }
+
+        await localStore.updateAccesses(
```

Comment:
for now I would still add a save within the updateAccess method

Reply:
done in https://github.com/wireapp/wire-ios/pull/2028/commits/dbcbbbc31b3d2c9fc17491832226eefaaf98e7f6